### PR TITLE
few updates to the base installer script

### PR DESCRIPTION
**Pull request by SpudGunMan**
little helper notice to unplug USB

### CODE REVIEW

**Comment by antonvh** [comment]
Thanks! I will have a look at it tonight.

**Comment by SpudGunMan** [comment]
there is random hash failure that was not trapped before, I haven't looked to see why the file hashes on Ctrl_plus and husky libs commonly fail a hash check, I will have it work fine once then fail for like a dozen times then work again fine. the chunking seems ok and the file is written to flash on the hub and all...

**Comment by SpudGunMan** [comment]
your installer was nice and I ported it out into its own to manage multiple projects feel free to take these files you can just drop off the 3rd party projects, I was going to add a function to copy binary files as well for support...
https://github.com/SpudGunMan/SpudGunMan/tree/main/Lego/legolibs

**Review by antonvh — APPROVED** [review]

**Review by antonvh — COMMENTED** [review]
Did you have trouble with this? If a file exists python just overwrites it, no need to remove the old one. Why did you add this?

File: Installer/create_install_script.py [hunk 1]
```diff
@@ -1,42 +1,70 @@
...
+#
+# Open the console/Debug and watch for notice to unplug USB BEFORE you upload/run
 
 import binascii, mpy_cross, time
 import hashlib
 import os
 from functools import partial
 
 LIB = '../mpy_robot_tools/'
+LIB_UART = '../../LMS-uart-esp/Libraries/UartRemote/MicroPython/uartremote.py'
```
Comment (antonvh) [inline Installer/create_install_script.py]: Please remove this. I would like to do this with a git submodule. That's more reliable.

**Review by antonvh — CHANGES_REQUESTED** [review]
I like the BAD_NAMES solution. I also like the renaming of the encoded variable. I like the readme update. But the lib should be solved differently. In the future it is better to create a different commit for every change. For now, can you do a new PR with only the changes that are ok? I will add the uartremote lib myself.

**Review by antonvh — CHANGES_REQUESTED** [review]
Great, thanks. Except for sleep_ms -> sleep. I don't see the point?

**Comment by antonvh** [comment]
> your installer was nice and I ported it out into its own to manage multiple projects feel free to take these files you can just drop off the 3rd party projects, I was going to add a function to copy binary files as well for support...

It would be nice if we could solve this in a general way. The installer could be a drop-in git submodule that scans a library folder for files and generates the scripts.

**Comment by SpudGunMan** [comment]
Hey I agree, this was a bit of a mess I was just trying to port up your script so I could quickly rebuild a stack of bricks without having to maintain updates etc.. 

 added the timeouts which seemed to help with some hash calculation issues. 

The fork I did was really my end goal and it now compiles all your stuff into a single directory and if its interesting to you I would take it ... also as you mentioned might just be worth putting all libs in one place so when you add more libs to it as you mentioned it will just compile them all..

**Comment by SpudGunMan** [comment]
not sure if you want to consider this fork, I sort of made a mess with starting edits to your installer here when it works just fine for your stuff and there is no real reason to take any of these changes check out..
https://github.com/SpudGunMan/SpudGunMan/blob/main/Lego/legolibs/create_install_script.py

**Comment by SpudGunMan** [comment]
to fully answer the why did I add a time.wait. was because I was having error I didn't trap yet.

I was following symlinks on osx/alias. it was a messy way to error check. I did away with following links as jumping project to project with them isn't working on my end anyway...

**Comment by antonvh** [comment]
If you don't mind, I'll cherry-pick some of your changes and move the whole installer to a drop-in submodule, sometime this week.

**Pull request closed without merging**